File: DeepBriscaTraining/briscola_gym/game_rules.py
values_points = {1: 11, 2: 0, 3: 10, 4: 0, 5: 0, 6: 0, 7: 0, 8: 2, 9: 3, 10: 4}


def select_winner(table: list, briscola):
    first = table[0]
    second = table[1]
    first_points = values_points[first.value]
    second_points = values_points[second.value]
    first_briscola = first.seed == briscola.seed
    second_briscola = second.seed == briscola.seed
    if second.seed == first.seed:
        if first_points == 0 and second_points == 0:
            return second.value > first.value
        else:
            return second_points > first_points
    elif first_briscola:
        return 0
    elif second_briscola:
        return 1
    else:
        return 0

File: DeepBriscaApp/lib/brisca_logic/game_rules.dart
import 'package:DeepBriscaApp/brisca_logic/card.dart';
import 'package:DeepBriscaApp/brisca_logic/constants.dart';

bool selectWinner(List<Card> table, Card briscola) {
  final first = table[0];
  final second = table[1];
  final firstPoints = valuesPoints[first.value] ?? 0;
  final secondPoints = valuesPoints[second.value] ?? 0;
  final secondBriscola = second.seed == briscola.seed;

  if (second.seed == first.seed) {
    if (firstPoints == 0 && secondPoints == 0) {
      return second.value < first.value;
    } else {
      return secondPoints < firstPoints;
    }
  } else if (secondBriscola) {
    return false;
  } else {
    return true;
  }
}